src/tests/functional/form_submission_tests.ts

```diff
@@ -24,7 +24,7 @@ export class FormSubmissionTests extends TurboDriveTestCase {
 
     this.assert.notOk(await this.formSubmitted)
     this.assert.equal(await this.pathname, "/src/tests/fixtures/one.html")
-    this.assert.equal(await this.visitAction, "advance")
+    this.assert.equal(await this.visitAction, "replace")
```

Comment:
Is `replace` the expected action for GET form submissions? 

It looks like we'll want a new test to validate normal visit redirects result in a new visit proposal with a `replace` action.

Reply:
The sequence:
- Visit the original URL with advance action
- If there is a redirection, propose replace visit to the destination with the existing response. That way it doesn't fire one more fetch call, instead reuses the existing response.

User will see the same behaviour as before. Just that programmatically the sequence looks like `advance` + `replace` instead of just `advance`.

As long as there is a redirect, `replace` action is expected at the end. In this test, form is submitted to `/__turbo/redirect` (ref: https://github.com/jaysson/turbo/blob/replace-on-redirect/src/tests/fixtures/form.html#L41), and that endpoint returns a 303 to `/src/tests/fixtures/one.html` resulting in `replace` action.